src/wp-includes/formatting.php

```diff
@@ -5978,8 +5978,21 @@ function _print_emoji_detection_script() {
 	}
 
 	wp_print_inline_script_tag(
-		sprintf( 'window._wpemojiSettings = %s;', wp_json_encode( $settings ) ) . "\n" .
-			file_get_contents( ABSPATH . WPINC . '/js/wp-emoji-loader' . wp_scripts_get_suffix() . '.js' )
+		wp_json_encode( $settings, JSON_HEX_TAG | JSON_UNESCAPED_SLASHES ),
+		array(
+			'id'   => 'wp-emoji-settings',
+			'type' => 'application/json',
+		)
+	);
+
+	$emoji_loader_script_path = '/js/wp-emoji-loader' . wp_scripts_get_suffix() . '.js';
+	wp_print_inline_script_tag(
+		rtrim( file_get_contents( ABSPATH . WPINC . $emoji_loader_script_path ) ) . "\n" .
+		'//# sourceURL=' . includes_url( $emoji_loader_script_path ),
+		array(
+			'type'  => 'module',
+			'async' => true,
```

Comment:
I considered making this async, but is that potentially harmful for performance? That will make it start to evaluate earlier, potentially competing with other things like classic scripts.

Is there a benefit to having it evaluate earlier?

Reply:
The benefit is it allows the worker to execute in parallel while the HTML is being parsed. I'll re-test.